src/backend/aspen/workflows/nextstrain_run/build_config.py

```diff
@@ -1,182 +1,76 @@
-
-        # Increase int'l sequences for state/country builds.
-        if (
-            self.tree_build_level != "location"
-            and subsampling["international"]["max_sequences"] < 100
-        ):
-            subsampling["international"]["max_sequences"] = 100
-
-
-def update_subsampling_for_location(tree_build_level, subsampling):
-    if tree_build_level == "country":
-        update_subsampling_for_country(subsampling)
-    if tree_build_level == "division":
-        update_subsampling_for_division(subsampling)
-
-
-def update_subsampling_for_country(subsampling):
-    # State and country aren't useful
-    if "state" in subsampling:
-        del subsampling["state"]
-    if "country" in subsampling:
-        del subsampling["country"]
-    # Update our local group query
-    subsampling["group"]["query"] = '''--query "(country == '{country}')"'''
-
-
-def update_subsampling_for_division(subsampling):
-    # State isn't useful
-    if "state" in subsampling:
-        del subsampling["state"]
-    # Update our local group query
-    subsampling["group"][
-        "query"
-    ] = '''--query "(division == '{division}') & (country == '{country}')"'''  # Keep the country filter in case of multiple divisions worldwide
-
-
-def apply_filters(config, subsampling, template_args):
-    filter_map = {"filter_start_date": "min_date", "filter_end_date": "max_date"}
-    for filter_name, yaml_key in filter_map.items():
-        value = template_args.get(filter_name)
-        if not value:
-            continue  # This filter isn't set, skip it.
-        # Support date expressions like "5 days ago" in our cron schedule.
-        value = dateparser.parse(value).strftime("%Y-%m-%d")
-        subsampling["group"][
-            yaml_key
-        ] = f"--{yaml_key.replace('_', '-')} {value}"  # ex: --max-date 2020-01-01
-
-    pango_lineages = template_args.get("filter_pango_lineages")
-    if pango_lineages:
-        # Nextstrain is rather particular about the acceptable syntax for
-        # values in the pango_lineages key. Before modifying please see
-        # https://discussion.nextstrain.org/t/failure-when-specifying-multiple-pango-lineages-in-a-build/670
-        clean_values = [re.sub(r"[^0-9a-zA-Z.]", "", item) for item in pango_lineages]
-        config["builds"]["aspen"]["pango_lineage"] = clean_values
-        # Remove the last " from our old query so we can inject more filters
-        old_query = subsampling["group"]["query"][:-1]
-        pango_query = " & (pango_lineage in {pango_lineage})"
-        subsampling["group"]["query"] = old_query + pango_query + '"'
+import yaml
+
+from aspen.database.models import Group, Pathogen
+from aspen.workflows.nextstrain_run.build_plugins.pathogen_plugins import (
+    MPXPlugin,
+    SC2Plugin,
+)
+from aspen.workflows.nextstrain_run.build_plugins.type_plugins import (
+    NonContextualizedPlugin,
+    OverviewPlugin,
+    TargetedPlugin,
+    TreeType,
+)
+
+
+class TemplateBuilder:
```

Comment:
Instead of using a factory that produces a Contextual/Overview/Whatever tree, we'll use a wrapper class that manages a set of config-modification plugins.